app/controllers/news_controller.rb

```diff
@@ -3,10 +3,20 @@ class NewsController < ApplicationController
   load_permissions_and_authorize_resource
 
   def index
-    @news = News.by_date
+    @news = @news.by_date
+    if params[:category].present?
+      @news = category(@news)
+    end
   end
 
   def show
     @news = News.find(params[:id])
   end
+
+  private
+
+  def category(news)
+    @category = Category.find_by_slug(params[:category])
```

Comment:
I would keep an eye on this select. Polymorphic relations tend to generate some funky SQL. If it does something really stupid you can always optimise the query later.